server/admin/webserver/webserver.go

```diff
@@ -0,0 +1,164 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package webserver
+
+import (
+	"context"
+	"crypto/sha256"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/go-chi/chi"
+	"github.com/go-chi/chi/middleware"
+
+	dexsrv "decred.org/dcrdex/server/dex"
+	"github.com/decred/slog"
+)
+
+const (
+	// rpcTimeoutSeconds is the number of seconds a connection to the
+	// RPC server is allowed to stay open without authenticating before it
+	// is closed.
+	rpcTimeoutSeconds = 10
+)
+
+var (
+	// Check that *dexsrv.DEX satifies ServerCore.
+	_   ServerCore = (*dexsrv.DEX)(nil)
+	log slog.Logger
+)
+
+// serverCore is satisfied by core.Core.
+type ServerCore interface{}
+
+// WebServer is a single-client http and websocket server enabling a browser
+// interface to the DEX client.
+type WebServer struct {
+	core      ServerCore
+	addr      string
+	tlsConfig *tls.Config
+	srv       *http.Server
+	authsha   [32]byte
+}
+
+// Config holds variables neede to create a new RPC Server.
+type Config struct {
+	Core                  ServerCore
+	Addr, Pass, Cert, Key string
+}
+
+// UseLogger sets the logger for the RPCServer package.
+func UseLogger(logger slog.Logger) {
+	log = logger
+}
+
+// filesExists reports whether the named file or directory exists.
+func fileExists(name string) bool {
+	_, err := os.Stat(name)
+	return !os.IsNotExist(err)
+}
+
+// New is the constructor for a new WebServer.
+func New(cfg *Config) (*WebServer, error) {
+	// Create authsha to verify requests against.
+	if cfg.Pass == "" {
+		return nil, fmt.Errorf("admin webserver password must be set to use the admin webserver")
+	}
+	authsha := sha256.Sum256([]byte(cfg.Pass))
```

Comment:
I hadn't thought about basic auth for this, but I'm not crazy about having the password stored in plaintext (in the configuration file).

Reply:
No longer stored. If the server is turned on the operator is prompted to enter a password.

Reply:
I understand the concern, but I'm also not wild about having `dcrdex` prompt for a password on startup.  However, when the dex private key is stored using the `encrypt` package, I suppose a password will be required on startup anyway.  I guess this prompting mechanism can serve both purposes.  BTW, is `ssh/terminal` how dcrwallet works with the `--promptpass` switch?

Reply:
yes
https://github.com/decred/dcrwallet/blob/bfe5c0810173856b263a9471128507b0aab9574f/internal/prompt/prompt.go#L20